index.js

```diff
@@ -110,68 +112,48 @@ const allFederalHolidaysForYear = (
-  if (shiftSaturdayHolidays || shiftSundayHolidays) {
-    holidays.forEach(holiday => {
-      // Allow the holiday objects to be modified inside this loop:
-      /* eslint-disable no-param-reassign */
-
-      const dow = holiday.date.getUTCDay();
-
-      if (dow === 0 && shiftSundayHolidays) {
-        // Actual holiday falls on Sunday. Shift the observed date forward to
-        // Monday.
-        holiday.date = new Date(
-          Date.UTC(
-            holiday.date.getUTCFullYear(),
-            holiday.date.getUTCMonth(),
-            holiday.date.getUTCDate() + 1
-          )
-        );
-      } else if (dow === 6 && shiftSaturdayHolidays) {
-        // Actual holiday falls on Saturday. Shift the observed date backward
-        // to Friday.
-        holiday.date = new Date(
-          Date.UTC(
-            holiday.date.getUTCFullYear(),
-            holiday.date.getUTCMonth(),
-            holiday.date.getUTCDate() - 1
-          )
-        );
-      }
-    });
-  }
-
-  holidays.forEach(holiday => {
-    holiday.dateString = `${holiday.date.getUTCFullYear()}-${holiday.date.getUTCMonth() +
-      1}-${holiday.date.getUTCDate()}`;
+  return holidays.map(holiday => {
+    let date = dayjs(holiday.date);
+
+    if (date.day() === 0 && shiftSundayHolidays) {
+      // Actual holiday falls on Sunday. Shift the observed date forward to
+      // Monday.
+      date = date.add(1, "day");
+    }
+
+    if (date.day() === 6 && shiftSaturdayHolidays) {
+      // Actual holiday falls on Saturday. Shift the observed date backward
+      // to Friday.
+      date = date.subtract(1, "day");
+    }
+
+    return {
+      name: holiday.name,
+      date: date.toDate(),
+      dateString: date.format("YYYY-MM-DD")
+    };
   });
-
-  return holidays;
 };
 
 const isAHoliday = (
   date = new Date(),
   { shiftSaturdayHolidays = true, shiftSundayHolidays = true, utc = false } = {}
 ) => {
-  const year = utc ? date.getUTCFullYear() : date.getFullYear();
+  const newDate = utc ? dayjs.utc(date) : dayjs(date);
+  const year = newDate.year();
+
   const shift = { shiftSaturdayHolidays, shiftSundayHolidays };
 
   // Get the holidays this year, plus check if New Year's Day of next year is
   // observed on December 31 and if so, add it to this year's list.
   const allForYear = allFederalHolidaysForYear(year, shift);
   const nextYear = allFederalHolidaysForYear(year + 1, shift);
-  if (nextYear[0].date.getUTCFullYear() === year) {
-    allForYear.push(nextYear[0]);
-  }
-
-  const mm = utc ? date.getUTCMonth() : date.getMonth();
-  const dd = utc ? date.getUTCDate() : date.getDate();
+  allForYear.push(nextYear[0]);
 
   // If any dates in this year's holiday list match the one passed in, then
   // the passed-in date is a holiday.  Otherwise, it is not.
   return allForYear.some(
-    holiday =>
-      holiday.date.getUTCMonth() === mm && holiday.date.getUTCDate() === dd
+    holiday => holiday.date.toDateString() === newDate.toDate().toDateString()
```

Comment:
```suggestion
    holiday => holiday.dateString === newDate.format("YYYY-MM-DD")
```

For timezones west of UTC and east of the international date line (mostly the Americas), all of the UTC holiday tests fail, because `toDateString` prints its results in your machine local time. So, for example, when given `31 May 2021` at midnight UTC, then `toDateString` will result in `Sun 30 Mar 2021`, which is not right. We also can't switch to `toISOString()` because then the local times will include a timezone offset while the UTC times do not.

Since the `dateString` string stored in the holiday objects does not account for time at all, we can use that instead.

But this makes me think I need to figure out how to run the CI tests in multiple timezones. 🤔